File: backend/BankLoanSimulator.Domain/Entities/Loan.cs
using BankLoanSimulator.Domain.Enums;

namespace BankLoanSimulator.Domain.Entitys
{
    public class Loan
    {
        public Guid Id { get; set; }
        public decimal Amount { get; set; }
        public decimal InterestRate { get; set; }
        public int TermInMonths { get; set; }
        public decimal MonthlyPayment { get; set; }
        public LoanStatusEnum Status { get; set; } = LoanStatusEnum.Pending;
        public DateTime RequestDate { get; set; } = DateTime.UtcNow;
        public DateTime? ReviewDate { get; set; }
        public string? AdminComments { get; set; }
        public Guid UserId { get; set; }
        public virtual User User { get; set; } = null!;
    }
}

File: frontend/bank-loan-simulator-ui/src/types/Loan.ts
export enum LoanStatus {
  Pending = 0,
  Approved = 1,
  Rejected = 2,
}

export interface Loan {
  id: string;
  amount: number;
  interestRate: number;
  termInMonths: number;
  monthlyPayment: number;
  status: LoanStatus;
  requestDate: string;
  reviewDate?: string;
  adminComments?: string;
  userId: string;
  user?: {
    fullName: string;
    email: string;
  };
}

export interface CreateLoanRequest {
  amount: number;
  interestRate: number;
  termInMonths: number;
}

export interface ReviewLoanRequest {
  status: LoanStatus;
  adminComments?: string;
}

export interface CalculatePaymentRequest {
  amount: number;
  interestRate: number;
  termInMonths: number;
}

export interface CalculatePaymentResponse {
  monthlyPayment: number;
}
